src/Nethermind/Nethermind.Synchronization/SnapSync/ProgressTracker.cs

```diff
@@ -471,5 +542,55 @@ public void Dispose()
                 range?.Dispose();
             }
         }
+
+        public interface IStorageRangeLock
+        {
+            IStorageRangeLock Increment(uint value = 1);
+            void Decrement(ValueHash256 key, bool removeFromOwner);
+            void ExecuteSafe(Action action);
+        }
+
+        public class StorageRangeLock(uint counter, ConcurrentDictionary<ValueHash256, IStorageRangeLock> owner)
```

Comment:
I'd just use a sharded lock to avoid such complications.
Also, you can just use `_pivot.UpdatedStorages.Add(pathWithAccount.Path);` to the splitted storage so that healing will fix it. But it limits the number of storage that can be splitted though.